Sources/Testing/ABI/ABI.swift

```diff
@@ -12,16 +12,68 @@
 @_spi(ForToolsIntegrationOnly)
 public enum ABI: Sendable {}
 
-// MARK: -
+// MARK: - ABI version abstraction
+
+extension ABI {
+  /// A protocol describing the types that represent different ABI versions.
+  protocol Version: Sendable {
+    /// The numeric representation of this ABI version.
+    static var versionNumber: Int { get }
+
+    /// Create an event handler that encodes events as JSON and forwards them to
+    /// an ABI-friendly event handler.
+    ///
+    /// - Parameters:
+    ///   - encodeAsJSONLines: Whether or not to ensure JSON passed to
+    ///     `eventHandler` is encoded as JSON Lines (i.e. that it does not
+    ///     contain extra newlines.)
+    ///   - eventHandler: The event handler to forward events to.
+    ///
+    /// - Returns: An event handler.
+    ///
+    /// The resulting event handler outputs data as JSON. For each event handled
+    /// by the resulting event handler, a JSON object representing it and its
+    /// associated context is created and is passed to `eventHandler`.
+    static func eventHandler(
+      encodeAsJSONLines: Bool,
+      forwardingTo eventHandler: @escaping @Sendable (_ recordJSON: UnsafeRawBufferPointer) -> Void
+    ) -> Event.Handler
+  }
+
+  /// The current supported ABI version (ignoring any experimental versions.)
+  static var currentVersion: (some Version).Type {
+    v0.self
+  }
```

Comment:
Could this be accompanied by a `typealias` e.g.
```swift
extension ABI {
  typealias CurrentVersion = v0
}
```
Then, any places that wish to reference the current version _as a type_ could more easily do so without needing to hardcode a specific version, e.g. `try JSON.decode(ABI.Record<ABI.CurrentVersion>.self, from: recordJSON)`